pydantic/types.py

```diff
@@ -243,6 +244,21 @@ def validate(cls, value: str) -> str:
         return validate_email(value)[1]
 
 
+class RelaxedBool(int):
```

Comment:
I think the only reason to use it would be performance; right now if you want speed when you *know* you are getting the right type, it's awkward. But I agree it would be better if the name reflected this.

I'd be okay either dropping or giving a different name.

Reply:
okay, let's keep it but say that in the docs.

Reply:
@samuelcolvin I did some minor investigation of the timing and I think the speed difference between the two bool validators is negligible, possibly with the standard one actually being faster than the relaxed one.

I think probably the only practical benefit is that you'd get to use a custom `__bool__` implementation if you wanted to use an arbitrary type in a field.

So I think the only thing to say in the docs is that it follows built-in python logic for boolean casting (what they currently say). Let me know if you still disagree, or if this changes your mind back to wanting to remove `RelaxedBool`.

Reply:
I would say let's remove it then. Nothing to stop people implementing it themselves if they really want it. But we we think there's basically no context in which you should use it, it shouldn't be added.

Sorry to flip flop.